### DISCUSSION TITLE
[Extension Development Host] and using WSL inside of it

### DISCUSSION THREAD

**Discussion by adriancostin6** [microsoft/vscode-discussions#2448] [Extension Development QnA] [answered]
Hey guys. I am developing an extension that wraps the C/C++ tools one to provide additional functionality, I will further refer to it as Extension-A for easier reference below. 

Whilst fixing a bug that is WSL specific, I am trying to test by starting the extension development host using the default debug configuration and then entering a WSL workspace via the dedicated extension for WSL. 

What I am running into is that when I connect to the WSL workspace inside of the extension development host, I seem to lose the debug context I had previously (access to Extension-A commands, settings and so on). 

This is a bummer, as I cannot debug my extension.  Is this unsuported, or am I obviously doing something wrong?

Configuration for debugging my extension is: 

```json
// A launch configuration that compiles the extension and then opens it inside a new window
// Use IntelliSense to learn about possible attributes.
// Hover to view descriptions of existing attributes.
// For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
{
        "version": "0.2.0",
        "configurations": [
                {
                        "name": "Run Extension",
                        "type": "extensionHost",
                        "request": "launch",
                        "args": [
                                "--extensionDevelopmentPath=${workspaceFolder}"
                        ],
                        "outFiles": [
                                "${workspaceFolder}/out/**/*.js"
                        ],
                        "preLaunchTask": "${defaultBuildTask}"
                },
                {
                        "name": "Extension Tests",
                        "type": "extensionHost",
                        "request": "launch",
                        "args": [
                                "--extensionDevelopmentPath=${workspaceFolder}",
                                "--extensionTestsPath=${workspaceFolder}/out/test/suite/index"
                        ],
                        "outFiles": [
                                "${workspaceFolder}/out/test/**/*.js"
                        ],
                        "preLaunchTask": "${defaultBuildTask}"
                }
        ]
}
```

**Comment by olguzzar** [comment, marked as answer]
Hi @adriancostin6, have you seen the [debugging using WSL extension docs](https://code.visualstudio.com/api/advanced-topics/remote-extensions#debugging-using-wsl)?

**Reply by adriancostin6** [reply]
Thanks for sharing. Makes a lot more sense now. I need my actual extension source in WSL to debug properly. I did not do that, only had local sources. Makes sense to be honest.

**Reply by olguzzar** [reply]
Great, so glad it was helpful!
